### PR TITLE
Exclude snakeyaml from pinot dependencies

### PR DESCRIPTION
**Pull request by wendigo**
It's used transitively by calcite to load model definitions

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-pinot/pom.xml [hunk 1]
```diff
@@ -324,6 +324,10 @@
                     <groupId>org.glassfish.jersey.core</groupId>
                     <artifactId>jersey-server</artifactId>
                 </exclusion>
+                <exclusion>
```
Comment (kokosing) [inline plugin/trino-pinot/pom.xml:327]: Add comment

**Review by kokosing — APPROVED** [review]

**Comment by martint** [comment]
What’s the motivation for this?

**Comment by wendigo** [comment]
@martint snakeyaml has known vulnerabilites that are reportered by security scanners. We don't use it so it's better to exclude it

**Comment by martint** [comment]
If it’s not used then those vulnerabilities don’t affect Trino, so what problem does it solve?

**Comment by wendigo** [comment]
@martint Some security scanners are eager to report vulnerability by mere existence of the jar. This particular CVE has score of 9.8/10

**Comment by martint** [comment]
Adding an exclusion is brittle. We need to remember to remove it when the library is updated, it adds more cruft to the pom, etc.

If it’s a problem, it’d be better to get it updated upstream.

**Comment by wendigo** [comment]
@martint This is harder to do and will take a lot more time and resources

**Comment by kokosing** [comment]
> Adding an exclusion is brittle

It is is. The library can be added again as transitive dependency for something else.

Let's use something like https://maven.apache.org/enforcer/enforcer-rules/bannedDependencies.html. Let's add it first here but then let's add it to airbase.

**Comment by wendigo** [comment]
@kokosing I don't know if we can do that. Tempto and benchto are still dependant on snakeyaml. I've updated it in https://github.com/trinodb/tempto/pull/109/files, but in benchto it's harder and will take some more time.

**Comment by kokosing** [comment]
We can do it as follow up.

File: testing/trino-product-tests-launcher/pom.xml [hunk 2]
```diff
@@ -71,16 +71,6 @@
             <artifactId>jackson-annotations</artifactId>
```
Comment (kokosing) [inline testing/trino-product-tests-launcher/pom.xml:71]: Can we move this commit out of this PR?

**Review by kokosing — APPROVED** [review]

**Review by kokosing — APPROVED** [review]

**Comment by hashhar** [comment]
maven-checks timeout is known issue - see https://github.com/trinodb/trino/pull/16863

**Comment by wendigo** [comment]
@hashhar cancelled :(

**Review by hashhar — APPROVED** [review]

**Pull request merged**